src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -532,12 +532,18 @@ public void onThrowable(final Throwable t) {
         public T onCompleted(final Response response) throws Exception {
             if (isSuccess(response)) {
                 final T value = MAPPER.readValue(response.getResponseBody(), clazz);
+
+                ((AbstractResponse) value).setStatusCode(response.getStatusCode());
+                ((AbstractResponse) value).setHeaders(response.getHeaders());
+
                 guavaFut.set(value);
                 return value;
             }
             else {
                 final ErrorResponse error = MAPPER.readValue(response.getResponseBody(), ErrorResponse.class);
                 final LobApiException exception = new LobApiException(response.getUri(), error);
+                exception.setStatusCode(response.getStatusCode());
```

Comment:
Couldn't there possibly be an exception that doesn't have a status code (like a DNS issue, etc). If that's the case, should we make `statusCode` an `Integer` rather than a `int` so it could be nullable? Do you know what `response.getStatusCode()` returns?